### PR TITLE
docs: add documentation for NegotiateAppVersion in apps.md

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

This PR adds documentation for `NegotiateAppVersion` in the applications section of the docs.

I have used the good explanation of @colin-axner in [this comment](https://github.com/cosmos/ibc-go/issues/136#issuecomment-897512989) as source.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by crodriguezvega** [comment]
Questions:
- Should I also add documentation for `NegotiateAppVersion` in the middleware section of the docs?
- Do we need to update the ICS030 to add the `NegotiateAppVersion` function as well there?

**Comment by colin-axner** [comment]
> Should I also add documentation for NegotiateAppVersion in the middleware section of the docs?

Yes most likely. Might be best to wait for ICS29 to support this function (if it doesn't already)

> Do we need to update the ICS030 to add the NegotiateAppVersion function as well there?

This might be useful, but `NegotiateAppVersion` is not an ICS requirement

File: docs/ibc/apps.md [hunk 1]
```diff
@@ -160,6 +160,36 @@ Application modules are expected to verify versioning used during the channel ha
 * `ChanOpenTry` callback should verify that the `MsgChanOpenTry.Version` is valid and that `MsgChanOpenTry.CounterpartyVersion` is valid.
 * `ChanOpenAck` callback should verify that the `MsgChanOpenAck.CounterpartyVersion` is valid and supported.
 
+IBC expects modules to implement the function `NegotiateAppVersion` from the application module
+interface. This function performs application version negotiation and returns the negotiated version.
```
Comment (damiannolan) [inline docs/ibc/apps.md:164]: Small nit here, [usually "method" is used](https://go.dev/doc/effective_go#interface_methods) when referring to a function that exists on an interface. And changed "application module" to [`IBCModule`](https://github.com/cosmos/ibc-go/blob/main/modules/core/05-port/types/module.go#L13)

```suggestion
IBC expects application modules to implement the `NegotiateAppVersion` method from the `IBCModule`
interface. This method performs application version negotiation and returns the negotiated version.
```

**Review by damiannolan — APPROVED** [review]

**Review by seantking — APPROVED** [review]
Thanks :handshake:

**Pull request merged**